Helper/Data.php

```diff
@@ -431,25 +389,131 @@ public function cronAbandonedcart($orders)
         }
     }
 
-    private function alertCustomer($row, $fields)
+    /**
+     * Formats cart fields to standard abandoned cart template fields used in Smaily.
+     *
+     * @param array $row            Cart information.
+     * @param array $selectedFields Fields selected for template.
+     * @return array                Array of 'customer_data' and 'products_data' sections.
+     */
+    private function prepareCartData($row, $selectedFields)
     {
-        $responderProduct = [];
+        // Populate customer data section.
+        $customerData = [
+            'email' => $row['customer_email']
+        ];
 
+        if (in_array('first_name', $selectedFields, true)) {
+            $customerData['first_name'] = $row['customer_firstname'];
+        }
+
+        if (in_array('last_name', $selectedFields, true)) {
+            $customerData['last_name'] = $row['customer_lastname'];
+        }
+        // Populate product data section.
+        $productsData = [];
+        $objectManager = \Magento\Framework\App\ObjectManager::getInstance();
+        $fieldsAvailable = $this->getAllAbandonedCartFields();
         foreach ($row['products'] as $product) {
             $_product = [];
-            foreach ($product as $field => $val) {
-                if ($field === 'name' || in_array($field, $fields, true)) {
-                    $_product['product_' . $field] = $val;
+            foreach ($fieldsAvailable as $field) {
+                if (in_array($field, $selectedFields)) {
+                    switch ($field) {
+                        case 'first_name':
+                        case 'last_name':
+                            // Skip customer fields.
+                            break;
+                        case 'qty':
+                            // Transform qty to quantity.
+                            $_product['product_quantity'] = $product[$field];
+                            break;
+                        case 'description':
+                            $productObject = $objectManager->
+                                create('Magento\Catalog\Model\Product')->
+                                load($product['product_id']);
+                            $description = $productObject->getDescription();
+                            $_product['product_description'] = htmlentities($description);
+                            break;
+                        default:
+                            $_product['product_' . $field ] = $product[$field];
```

Comment:
Magento formats prices as follows:
```
'product_quantity' => '1.0000',
'product_price' => '34.0000',
'product_base_price' => '34.0000',
```

Should I format prices to two (three?) decimal places or is this a responsibility for the Smaily template building process? I know that these decimal places vary between integrations and platforms.

Reply:
Prices should be formatted as they are in store - `$34`, `€34` or `€34.00`. With product quantity maybe it is best if we format:
- `1.0000` as `1`;
- `1.2300` as `1.23`;
- `1.2345` as `1.2345`;

What do you think? Does it make sense?

Reply:
Ok, I will format the price to 2 decimal places to be consistent and the quantity as described before.

Reply:
Can you use the stores price formatting helper? Then you wouldn't need to worry about the formatting and let the store configuration handle it for you.

Reply:
If I go with the price formatting helper that Magento uses I also get currency symbol.
```
      'product_quantity' => '1',
      'product_price' => '34,00 €',
      'product_base_price' => '34,00 €',
```
If it is the preferred format we want to pass it is easily achievable. The question is if it is so easy with other plugins also. Thinking of for example if the `$` sign is used before a price or after. Or the delimiter if they use `,` or `.`

If we provide a single stock template for abandoned cart emails there might run into a collision there later where we get a standard field `product_price_1` but the format varies between integrations.

Reply:
Currently, Prestashop default uses it as follows:
```
'product_price_1' => '€34.46',
'product_price_2' => '€42.00',
```

Maybe it is not an issue if the use case is just to display a price.

Reply:
Now using Magneto built-in currency functions to pass product price as it is displayed in the store. https://github.com/sendsmaily/smaily-magento-extension/pull/33/commits/9dbdaa5eac83a04fae5e36206445be298ffa0609

Reply:
Excellent! I would rather see the same formatting for prices as it is in store, then the user wouldn't need to worry about formatting for different stores that use the same abandoned cart logic.